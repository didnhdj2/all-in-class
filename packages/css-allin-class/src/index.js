import { getPlugin } from './utils/plugins.js'

export default function createPlugin(userConfig = {}) {
	return getPlugin(userConfig)
}

export function allin(userConfig = {}) {
	return getPlugin(userConfig)
}