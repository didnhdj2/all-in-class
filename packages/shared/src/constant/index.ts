export const PLUGIN_NAME: string = 'plugin-prefixed-allin-class'
export const PLUGIN_PREFIX: string = 'ALLIN_CLASS'
export const POLYFILL_ID: string = 'allinclass.css'
export const CSS_PLACE_HOLDER_KEY: string = '#ALLIN_CSS_PLACE_HOLDER_KEY{color:red}'