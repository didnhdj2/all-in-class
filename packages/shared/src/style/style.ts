import { isArray, isFunction, isObject, isString } from "../index"

export function addFix(val: string, prefix: string = "", suffix: string = "") {
	return `${prefix}${val}${suffix}`
}

export function removeFix(val: string, prefix: string, suffix: string) {
	let resStr = val;
	if (prefix) {
		const pre = RegExp(`^${prefix}`, 'g');
		resStr = val.replace(pre, '')
	}
	if (suffix) {
		const suf = RegExp(`^${suffix}$`, 'g');
		resStr = val.replace(suf, '')
	}

	return resStr
}

interface UserConfig {
	prefix?: string;
	suffix?: string;
}

interface Preset {
	rules?: Array<any>;
}
interface Rules {
	[propname:string]: any;
}
/**
 * 用户classToken来匹配值
 * @param {object} preset {rules,....} 配置的规则
 * @param {object} userConfig 
 */

export function separateRules(preset: any, userConfig: UserConfig = {}) {
	let dynamicRules: Rules = {}
	let staticRules: Rules = {}
	const { prefix, suffix } = userConfig

	for (let rule of preset.rules) {
		if (isArray(rule) && rule[0] && rule[1] && isFunction(rule[1])) {
			// [rule,()=>{}]
			dynamicRules[rule[0]] = rule
			continue
		}

		if (isArray(rule) && rule[0] && rule[1] && isString(rule[0]) && isString(rule[1])) {
			let [key, val] = rule;
			// 添加前缀和后缀
			const prefixedKey:string = addFix(key, prefix, suffix)

			// {rule:value}
			staticRules[prefixedKey] = addSemicolon(val)
			continue
		}

		if (isObject(rule)) {
			for (let [key, val] of Object.entries(rule)) {
				if (isString(key) && isString(val)) {

					// 添加前缀和后缀
					const prefixedKey:string = addFix(key, prefix, suffix)

					// {rule:value}
					staticRules[prefixedKey] = addSemicolon(val as string)
				}
			}
			continue
		}

		// throw new Error(`非法的规则:${rule}`)
	}

	return { dynamicRules, staticRules }
}

function addSemicolon(rule: string) {
	return rule.endsWith(';') ? rule : rule + ';';
}
