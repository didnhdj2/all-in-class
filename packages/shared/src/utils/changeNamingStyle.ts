import { unique } from "../index";

export function isAllUpperCase(str: string) {
  return str === str.toUpperCase();
}

// 将字符串转为小驼峰命名法（camelCase）
export function toCamelCase(str: string) {
	return str.replace(/[-_\.\s]+(.)?/g, (_, c) => c ? c.toUpperCase() : '');
}

// 将字符串转为大驼峰命名法（PascalCase）
export function toPascalCase(str: string) {
	return str.replace(/[-_\.\s]+(\w)/g, (_, c) => c.toUpperCase()).replace(/^\w/, c => c.toUpperCase());
}

// 将字符串转为下划线命名法（snake_case）
export function toSnakeCase(str: string) {
	return str.replace(/[-\.\s]/g, '_').toLowerCase();
}

// 将字符串转为下划线命名法（aa_bb_cc）
function upperToSnakeCase(str: string) {
  return str.replace(/[A-Z]/g, (match) => '_' + match.toLowerCase());
}

// 将字符串转为全大写形式
export function toUpperCase(str: string) {
	return str.toUpperCase();
}

// 将字符串转为全小写形式
export function toLowerCase(str: string) {
	return str.toLowerCase();
}

// 将字符串转为全大写形式
export function toPureUpperCase(str: string) {
	return str.replace(/[-_\.\s]+/g, '').toUpperCase();
}
// 将字符串转为全小写形式
export function toPureLowerCase(str: string) {
	return str.replace(/[-_\.\s]+/g, '').toLowerCase();
}

// 将字符串转换为下划线格式（所有单词小写，单词之间用下划线连接）
export function toUnderscoreCase(str: string) {
	return str.replace(/[-_\.\s]+/g, '_').toLowerCase();
}
// 将字符串转为中划线命名法（kebab-case）
export function toKebabCase(str: string) {
	return str.replace(/[-_\.\s]+/g, '-').toLowerCase();
}

export function transformCase(str: string) {
	return unique([
		toCamelCase(str),
		toPascalCase(str),
		toUnderscoreCase(str),
		toUnderscoreCase(str).toUpperCase(),
		toKebabCase(str),
		toKebabCase(str).toUpperCase(),
		toUpperCase(str),
		toLowerCase(str),
		toPureUpperCase(str),
		toPureLowerCase(str),
	])
}

// 把所有格式转换为空格格式
export function convertToWords(str: string) {
	if (isAllUpperCase(str)) {
		str = str.toLowerCase()
	}
  return str.replace(/[-\._\s]/g, ' ')  // 将中划线和下划线替换为空格
            .replace(/[a-z][A-Z]/g, function(match) {
              return match.charAt(0) + ' ' + match.charAt(1); // 在小驼峰和大驼峰的单词分隔点处插入空格
            })
            .replace(/[A-Z]+/g, function(match) {
              return ' ' + match.toLowerCase(); // 将大写字母单词转换为小写字母单词，并在前面插入空格
            })
            .replace(/\s+/g, ' ') // 去除多余的空格
            .trim(); // 去除字符串两端的空格
}