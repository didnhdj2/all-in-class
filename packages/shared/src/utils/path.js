import { dirname, resolve, join, isAbsolute, normalize } from 'path'

export function normalizeAbsolutePath(path) {
	if (isAbsolute(path)) {
		return normalize(path)
	}

	return path
}


