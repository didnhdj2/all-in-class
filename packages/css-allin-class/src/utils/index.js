import { PLUGIN_PREFIX } from "../constant"



const factory = (name) => {
	return (value) => Object.prototype.toString.call(value) === `[object ${name}]`
}
export const isNull = factory('Null')
export const isObject = factory('Object')
export const isArray = factory('Array')
export const isMap = factory('Map')
export const isSet = factory('Set')
export const isPromise = factory('Promise')
export const isAsyncFunction = factory('AsyncFunction')

export const isString = (value) => typeof value === 'string'
export const notNull = (value) => value !== null
export const isEmpty = (value) => value !== null || value !== undefined || value !== ''
export const isFunction = (value) =>  factory('Function')(value)||isAsyncFunction(value)

export function toArray(value = []) {
	return isArray(value) ? value : [value]
}

export const toNoRepeatArray = (value) => Array.from(new Set(value))

export function noop() {}

export function warn(msg) {
	console.warn(`[${PLUGIN_PREFIX}]`, msg)
}
export function error(msg) {
	console.error(`[${PLUGIN_PREFIX}]`, msg)
}

