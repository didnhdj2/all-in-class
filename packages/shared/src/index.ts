export * from './constant/index'
export * from './style/index'
export * from './utils/index'

import { PLUGIN_PREFIX } from "./constant/index"

const factory = (name: string): Function =>  (val: any): boolean  => Object.prototype.toString.call(val) === `[object ${name}]`
export const isMap = factory('Map')
export const isSet = factory('Set')
export const isAsyncFunction = factory('AsyncFunction')
export const isRegExp = factory('RegExp')
export const isDate = factory('Date')
export const isPlainObject = factory('Object')

export const isFunction = (val: any) => typeof val === 'function'
export const isString = (val: any) => typeof val === 'string'
export const isSymbol = (val: any) => typeof val === 'symbol'
export const isObject = (val: any) => val !== null && typeof val === 'object'
export const isEmpty = (value: any) => value === null || value === undefined || value === ''

export const isArray = Array.isArray
export const extend = Object.assign

export function isPromise(val: any) {
	// 根据 Promise/A+ 规范，一个对象被视为 Promise 对象，需要满足以下条件：
	// 该对象必须是一个对象或函数，
	return isObject(val) &&
		// 该对象必须具有 then 方法，其类型为函数。具有 catch 方法，其类型为函数。
		isFunction(val.then) && isFunction(val.catch) &&
		// 该对象必须具有一个状态属性，该状态属性只能是以下三种状态之一：pending（等待中）、fulfilled（已完成）或 rejected（已拒绝）。
		isString(val.status) && ['pending', 'fulfilled', 'rejected'].includes(val.status) &&
		// 该对象必须具有一个 thenable 属性，该属性必须是一个函数或对象，它的行为类似于 Promise 对象。
		(isObject(val.thenable) || isFunction(val.thenable))
}

const hasOwnProperty = Object.prototype.hasOwnProperty
export const hasOwnProp = (obj:object, key:string) => hasOwnProperty.call(obj, key)

export function toArray(value = []) {
	return isArray(value) ? value : [value]
}
export const unique = (array: Array<any>) => Array.from(new Set(array))

// export const toNoRepeatArray = (value) => Array.from(new Set(value))
export function noop() {}

// const warned = new Set()
// export function warnOnce(msg) {
//   if (warned.has(msg)) return
//   console.warn('[odcss]', msg)
//   warned.add(msg)
// }

export function warn(msg:string) {
	console.warn(`[${PLUGIN_PREFIX}]`, msg)
}
export function error(msg:string) {
	console.error(`[${PLUGIN_PREFIX}]`, msg)
}

export const remove = (arr: Array<any>, el:any) => {
	const i = arr.indexOf(el)
	if (i > -1) {
		arr.splice(i, 1)
	}
}



// export function mergeDeep<T>(original: T, patch: DeepPartial<T>): T {
//   const o = original as any
//   const p = patch as any

//   if (Array.isArray(o))
//     return [...p] as any

//   const output = { ...o }
//   if (isObject(o) && isObject(p)) {
//     Object.keys(p).forEach((key) => {
//       if (((isObject(o[key]) && isObject(p[key])) || (Array.isArray(o[key]) && Array.isArray(p[key]))))
//         output[key] = mergeDeep(o[key], p[key])
//       else
//         Object.assign(output, { [key]: p[key] })
//     })
//   }
//   return output
// }

// export function clone<T>(val: T): T {
//   let k: any, out: any, tmp: any

//   if (Array.isArray(val)) {
//     out = Array(k = val.length)
//     // eslint-disable-next-line no-cond-assign
//     while (k--) out[k] = ((tmp = val[k]) && typeof tmp === 'object') ? clone(tmp) : tmp
//     return out as any
//   }

//   if (Object.prototype.toString.call(val) === '[object Object]') {
//     out = {} // null
//     for (k in val) {
//       if (k === '__proto__') {
//         Object.defineProperty(out, k, {
//           value: clone((val as any)[k]),
//           configurable: true,
//           enumerable: true,
//           writable: true,
//         })
//       }
//       else {
//         // eslint-disable-next-line no-cond-assign
//         out[k] = ((tmp = (val as any)[k]) && typeof tmp === 'object') ? clone(tmp) : tmp
//       }
//     }
//     return out
//   }

//   return val
// }
