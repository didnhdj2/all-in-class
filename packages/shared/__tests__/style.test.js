import { describe, assert, expect, test, it } from 'vitest'
import { isEmpty, isFunction, unique } from '../src';
import { separateRules } from '../src/style/style';
import { transformCase, convertToWords } from '../src/utils/changeNamingStyle'
import { preset } from './data';


describe('预设规则', async () => {
	it('is', async ({ expect }) => {
		expect(separateRules({rules:preset})).toMatchSnapshot()
	})
})

