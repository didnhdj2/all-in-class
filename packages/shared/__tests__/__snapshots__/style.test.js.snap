// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`预设规则 > is 1`] = `
{
  "dynamicRules": {
    "/^(bottom|b)-(-?\\\\d*\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^\\(bottom\\|b\\)-\\(-\\?\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^(br|radius|rounded)(?:-(tr|tl|br|bl))?-?(\\\\d+)-?(\\\\d+)?-?(\\\\d+)?-?(\\\\d+)?(rem|em|px|rpx|%)?$/": [
      /\\^\\(br\\|radius\\|rounded\\)\\(\\?:-\\(tr\\|tl\\|br\\|bl\\)\\)\\?-\\?\\(\\\\d\\+\\)-\\?\\(\\\\d\\+\\)\\?-\\?\\(\\\\d\\+\\)\\?-\\?\\(\\\\d\\+\\)\\?\\(rem\\|em\\|px\\|rpx\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^(color|c)-#([\\\\d\\\\w]{1,6})$/": [
      /\\^\\(color\\|c\\)-#\\(\\[\\\\d\\\\w\\]\\{1,6\\}\\)\\$/,
      [Function],
    ],
    "/^(color|c)-(\\\\w+)$/": [
      /\\^\\(color\\|c\\)-\\(\\\\w\\+\\)\\$/,
      [Function],
    ],
    "/^(color|c)-(cur|inherit|initial|unset)$/": [
      /\\^\\(color\\|c\\)-\\(cur\\|inherit\\|initial\\|unset\\)\\$/,
      [Function],
    ],
    "/^(height|h)-(\\\\d*\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^\\(height\\|h\\)-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^(left|l)-(-?\\\\d*\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^\\(left\\|l\\)-\\(-\\?\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^(right|r)-(-?\\\\d*\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^\\(right\\|r\\)-\\(-\\?\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^(top|t)-(-?\\\\d*\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^\\(top\\|t\\)-\\(-\\?\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^(width|w)-(\\\\d*\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^\\(width\\|w\\)-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^bg-#([\\\\d\\\\w]+)$/": [
      /\\^bg-#\\(\\[\\\\d\\\\w\\]\\+\\)\\$/,
      [Function],
    ],
    "/^bg-(\\\\w+)$/": [
      /\\^bg-\\(\\\\w\\+\\)\\$/,
      [Function],
    ],
    "/^bg-linear-(\\\\d+)-(\\\\d+)-(\\\\d+)$/": [
      /\\^bg-linear-\\(\\\\d\\+\\)-\\(\\\\d\\+\\)-\\(\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^bg-rgba-(\\\\d+)-(\\\\d+)-(\\\\d+)-((\\\\d+)|(?:0*\\\\.(\\\\d+)))$/": [
      /\\^bg-rgba-\\(\\\\d\\+\\)-\\(\\\\d\\+\\)-\\(\\\\d\\+\\)-\\(\\(\\\\d\\+\\)\\|\\(\\?:0\\*\\\\\\.\\(\\\\d\\+\\)\\)\\)\\$/,
      [Function],
    ],
    "/^border-#([\\\\d\\\\w]+)$/": [
      /\\^border-#\\(\\[\\\\d\\\\w\\]\\+\\)\\$/,
      [Function],
    ],
    "/^border-?(t|l|r|b)?-(\\\\d+)(px|rpx)?$/": [
      /\\^border-\\?\\(t\\|l\\|r\\|b\\)\\?-\\(\\\\d\\+\\)\\(px\\|rpx\\)\\?\\$/,
      [Function],
    ],
    "/^f-(\\\\d+)$/": [
      /\\^f-\\(\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^fs-(\\\\d+)$/": [
      /\\^fs-\\(\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^fw-(\\\\d+)$/": [
      /\\^fw-\\(\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^gba-(\\\\d+)-(\\\\d+)-(\\\\d+)$/": [
      /\\^gba-\\(\\\\d\\+\\)-\\(\\\\d\\+\\)-\\(\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^inset-(-?0?\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^inset-\\(-\\?0\\?\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^lh-(\\\\d+|\\\\d*\\\\.\\\\d+)(rem|vh|vw|px|rpx|em)*$/": [
      /\\^lh-\\(\\\\d\\+\\|\\\\d\\*\\\\\\.\\\\d\\+\\)\\(rem\\|vh\\|vw\\|px\\|rpx\\|em\\)\\*\\$/,
      [Function],
    ],
    "/^line-(\\\\d+)$/": [
      /\\^line-\\(\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^m-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^m-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^m-(\\\\d*\\\\.?\\\\d+)-(\\\\d*\\\\.?\\\\d+)-(\\\\d*\\\\.?\\\\d+)-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^m-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^m-(\\\\d*\\\\.?\\\\d+)-(\\\\d*\\\\.?\\\\d+|auto)$/": [
      /\\^m-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\|auto\\)\\$/,
      [Function],
    ],
    "/^max-h-(\\\\d*\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^max-h-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^max-w-(\\\\d*\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^max-w-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^mb-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^mb-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^min-h-(\\\\d*\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^min-h-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^min-w-(\\\\d*\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^min-w-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^ml-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^ml-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^mlr-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^mlr-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^mr-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^mr-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^mt-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^mt-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^mtb-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^mtb-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^opacity-(0?\\\\.?\\\\d+)$/": [
      /\\^opacity-\\(0\\?\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^order-(-?\\\\d*\\\\.?\\\\d+)$/": [
      /\\^order-\\(-\\?\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^p-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^p-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^p-(\\\\d*\\\\.?\\\\d+)-(\\\\d*\\\\.?\\\\d+)-(\\\\d*\\\\.?\\\\d+)-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^p-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^p-(\\\\d*\\\\.?\\\\d+)-(\\\\d+|auto)$/": [
      /\\^p-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)-\\(\\\\d\\+\\|auto\\)\\$/,
      [Function],
    ],
    "/^pb-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^pb-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^pl-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^pl-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^plr-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^plr-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^pr-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^pr-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^pt-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^pt-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^ptb-(\\\\d*\\\\.?\\\\d+)$/": [
      /\\^ptb-\\(\\\\d\\*\\\\\\.\\?\\\\d\\+\\)\\$/,
      [Function],
    ],
    "/^rgba-(\\\\d+)-(\\\\d+)-(\\\\d+)-((\\\\d+)|(?:0*\\\\.(\\\\d+)))$/": [
      /\\^rgba-\\(\\\\d\\+\\)-\\(\\\\d\\+\\)-\\(\\\\d\\+\\)-\\(\\(\\\\d\\+\\)\\|\\(\\?:0\\*\\\\\\.\\(\\\\d\\+\\)\\)\\)\\$/,
      [Function],
    ],
    "/^shadow-?(\\\\d+)?-?(\\\\d+)?-?(\\\\d+)?-?(\\\\d+)?-?(0?\\\\.\\\\d+)?$/": [
      /\\^shadow-\\?\\(\\\\d\\+\\)\\?-\\?\\(\\\\d\\+\\)\\?-\\?\\(\\\\d\\+\\)\\?-\\?\\(\\\\d\\+\\)\\?-\\?\\(0\\?\\\\\\.\\\\d\\+\\)\\?\\$/,
      [Function],
    ],
    "/^translateX-(0?\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^translateX-\\(0\\?\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^translateY-(0?\\\\.?\\\\d+)(rem|vh|vw|%)?$/": [
      /\\^translateY-\\(0\\?\\\\\\.\\?\\\\d\\+\\)\\(rem\\|vh\\|vw\\|%\\)\\?\\$/,
      [Function],
    ],
    "/^z-(-?\\\\d+|auto)$/": [
      /\\^z-\\(-\\?\\\\d\\+\\|auto\\)\\$/,
      [Function],
    ],
  },
  "staticRules": {
    "absolute": "position: absolute;",
    "align-baseline": "vertical-align: baseline;",
    "align-bottom": "vertical-align: bottom;",
    "align-middle": "vertical-align: middle;",
    "align-text-bottom": "vertical-align: text-bottom;",
    "align-text-top": "vertical-align: text-top;",
    "align-top": "vertical-align: top;",
    "as-b": "align-self: baseline;",
    "as-c": "align-self: center;",
    "as-e": "align-self: flex-end;",
    "as-s": "align-self: flex-start;",
    "as-st": "align-self: stretch;",
    "block": "display: block;",
    "border-dashed": "border-style: dashed;",
    "border-dotted": "border-style: dotted;",
    "border-double": "border-style: double;",
    "border-none": "border-style: none;",
    "border-solid": "border-style: solid;",
    "border-transparent": "border-color: transparent;",
    "box-border": "	box-sizing: border-box;",
    "box-content": "box-sizing: content-box;",
    "capitalize": "text-transform: capitalize;",
    "f-ac": "align-items: center;",
    "f-ac-jc": " align-items: center;justify-content: center;",
    "f-ac-je": " align-items: center;justify-content: flex-end;",
    "f-ac-jsa": " align-items: center;justify-content: space-around;",
    "f-ac-jsb": " align-items: center;justify-content: space-between;",
    "f-ae": "align-items: flex-end;",
    "f-as": "align-items: flex-start;",
    "f-jc": "justify-content: center;",
    "f-je": "justify-content: flex-end;",
    "f-js": " justify-content: flex-start;",
    "f-jsa": "justify-content: space-around;",
    "f-jsb": "justify-content: space-between;",
    "fc": "display: flex;flex-direction: column;",
    "fixed": "position: fixed;",
    "flex": "display: flex;",
    "fr": "display: flex;flex-direction: row;",
    "fw": "flex-wrap:wrap;",
    "fw-b": "font-weight: bold;",
    "grid": "display: grid;",
    "inline-block": "display: inline-block;",
    "invisible": "visibility: hidden;",
    "isolate": "isolation: isolate;",
    "isolation-auto": "isolation: auto;",
    "line": " border-bottom: 2px solid #CDCDCD; width: 100%;height: 2px;",
    "line-dark": " border-bottom: 2px solid #797B89; width: 100%;height: 2px;",
    "line-through": "text-decoration: line-through;",
    "lowercase": "text-transform: lowercase;",
    "max-h-max": "max-height: max-content;",
    "max-h-min": "max-height: min-content;",
    "max-h-none": "max-height: none;",
    "max-w-max": "max-width: max-content;",
    "max-w-min": "max-width: min-content;",
    "max-w-none": "max-width: none;",
    "min-h-max": "min-height: max-content;",
    "min-h-min": "min-height: min-content;",
    "min-h-none": "min-height: none;",
    "min-w-max": "min-width: max-content;",
    "min-w-min": "min-width: min-content;",
    "min-w-none": "min-width: none;",
    "no-decoration": "text-decoration: none;",
    "none": "display: none;",
    "normal-case": "text-transform: none;",
    "poiter": "cursor: pointer;",
    "relative": "position: relative;",
    "scroll-x": " overflow-x: scroll;",
    "scroll-y": " overflow-y: scroll;",
    "static": "position: static;",
    "sticky": "position: sticky;",
    "td-lt": "text-decoration: line-through;",
    "text-c": "text-align: center;",
    "text-center": "text-align: center;",
    "text-j": "text-align: justify;text-align-last: justify;",
    "text-justify": "text-align: justify;",
    "text-l": "text-align: left;",
    "text-left": "text-align: left;",
    "text-r": "text-align: right;",
    "text-right": "text-align: right;",
    "un-line": "text-decoration: underline;",
    "underline": "text-decoration: underline;",
    "uppercase": "text-transform: uppercase;",
    "visible": "visibility: visible;",
  },
}
`;
